### ISSUE TITLE
Steam World Dig 2 (571310)

### ISSUE THREAD

**Issue by hasezoey** [ValveSoftware/Proton#6376]
# Compatibility Report
- Name of the game with compatibility issues: Steam World Dig 2
- Steam AppID of the game: 571310

## System Information
- GPU: AMD Vega 64
- Driver/LLVM version: AMD AMD Radeon RX Vega (vega10, LLVM 14.0.6, DRM 3.48, 6.0.11-1-MANJARO)
- Kernel version: 6.0.11
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/hasezoey/04987fc45b69e629829295250732c6d9
- Proton version: 1670512927 experimental-7.0-20221208

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-571310.log](https://github.com/ValveSoftware/Proton/files/10200255/steam-571310.log)

## Symptoms 

With Proton and no extra environment variables: `Could not initialized SDL 2, error: wayland not available`
With Proton and `SDL_VIDEODRIVER=x11`: `Could not initialized SDL 2, error: x11 not available`
Bare and with Steam-Linux-Runtime (solider): `Failed to initialized OpenGL extensions (GLEW), error: Unknown error`
Bare and with Steam-Linux-Runtime (solider) and `SDL_VIDEODRIVER=x11`: it runs without problems

## Reproduction

1. have a system with wayland (in my case kwin_wayland / KDE Plasma)
2. install game on steam
3. set to proton version (Experimental in my case)
4. try to launch the game

---

Edit: in case it matters, here is a list of the sdl packages installed on my maschine:

(output from manjaro `pacman -Ss sdl | grep installed`)
```txt
extra/qemu-audio-sdl 7.1.0-11 [installed]
extra/qemu-ui-sdl 7.1.0-11 [installed]
extra/sdl2 2.26.0-1 [installed]
extra/sdl2_image 2.6.2-2 [installed]
multilib/lib32-sdl2 2.24.2-1 [installed]
```

**Comment by kisak-valve** [comment]
Hello @hasezoey, you're setting `SDL_VIDEODRIVER=wayland` globally somewhere (https://gist.github.com/hasezoey/04987fc45b69e629829295250732c6d9#file-steaminfo-txt-L154) and that's interfering with SDL run inside Proton in a similar way as https://github.com/ValveSoftware/Proton/issues/5587#issuecomment-1221255936.

You could try setting the game's launch options to `SDL_VIDEODRIVER= %command%` or maybe `SDL_VIDEODRIVER=windows %command%` to override your session-wide setting. (https://wiki.libsdl.org/SDL2/FAQUsingSDL)

**Comment by hasezoey** [comment]
> Hello @hasezoey, you're setting SDL_VIDEODRIVER=wayland globally somewhere (https://gist.github.com/hasezoey/04987fc45b69e629829295250732c6d9#file-steaminfo-txt-L154) 

true, kind of forgot about that, though i was confused because the normal fallback i would do `=x11` also did not work. forgot that that sdl instance was for *windows* not *linux* and so did not have those protocols, and also didnt know `=windows` would be the option, thanks for mentioning that.

---

the workaround (for my case) which i ended up using was using `SDL_VIDEODRIVER=x11` on the native linux version of the game, which worked great - for that one i was just really confused because the error did not mention anything about SDL (for context i mean the error of `Failed to initialized OpenGL extensions (GLEW), error: Unknown error`)
